api/src/main/scala/Utils.scala

```diff
@@ -147,6 +147,23 @@ trait Utils {
   }
 
   /**
+    * Given a sequence of MultiLines and an area of interest, crops the lines
+    * to the area of interest and returns a sequence containing the cropped lines.
+    *
+    * @param   lines  A sequence of MultiLines
+    * @param   aoi    Area of Interest
+    * @return         A sequence of MultiLines that intersect with the Area of Interest
+    */
+  def cropLinesToAOI(lines: Seq[MultiLine], aoi: MultiPolygon): Seq[MultiLine] = {
+    lines.map(line => {
+      (line & aoi).asMultiLine match {
+        case Some(ml) => ml
+        case None => MultiLine.EMPTY
+      }
+    }).filter(line => line != MultiLine.EMPTY)
```

Comment:
Can this be a `flatMap` instead to filter out the `None` types in the first pass?